rust/main/chains/hyperlane-ethereum/src/tx.rs

```diff
@@ -219,20 +220,25 @@ where
 pub(crate) async fn call_with_lag<M, T>(
     call: ethers::contract::builders::ContractCall<M, T>,
     provider: &M,
-    maybe_lag: Option<NonZeroU64>,
+    lag: &ReorgPeriod,
 ) -> ChainResult<ethers::contract::builders::ContractCall<M, T>>
 where
     M: Middleware + 'static,
     T: Detokenize,
 {
-    if let Some(lag) = maybe_lag {
-        let fixed_block_number: BlockNumber = provider
-            .get_block_number()
-            .await
-            .map_err(ChainCommunicationError::from_other)?
-            .saturating_sub(lag.get().into())
-            .into();
-        Ok(call.block(fixed_block_number))
+    if !lag.is_none() {
+        let reorg_period = EthereumReorgPeriod::try_from(lag)?;
```

Comment:
it's a bit of a shame we need to do this conversion so often when most of the ethereum structs have a `reorg_period: EthereumReorgPeriod` readily available

I guess the main reason is that the chain agnostic `Mailbox` and `MerkleTreeHook` traits from `hyperlane-core` require the reorg period to be passed in for functions like `count` / `tree` / `latest_checkpoint` etc

Iirc we originally did this to underscore how important it was that those functions only use finalized state. @daniel-savu curious what you'd think of removing those params from the chain agnostic traits and requiring implementations to handle finality concerns internally?

I'm happy if we continue as it is like this, but something about this feels a bit weird

Reply:
while I agree we could push more encapsulation in the chain-specific implementations, I think the specific change you're suggesting would be quite invasive and would need to be part of a bigger chain builder overhaul. To me it's really the chain builders that feel weird rather than these conversions specifically :P